### PR TITLE
[WIP] Reimplement lambdamart ndcg.

### PR DESCRIPTION
**Pull request by trivialfis**
Related: https://github.com/dmlc/xgboost/issues/6450

* Add parameter `ndcg_truncation`.
* Change the default metric to `ndcg@truncation`.

TODOS:
- [ ] Better tests.
- [ ] Benchmark for accuracy and perf.
- [x] Label checks.
- [ ] Gain type.

### CODE REVIEW

**Comment by trivialfis** [comment]
Question to myself: Is there a way to use cub sort pair with discard iterator to avoid allocating an array?

File: python-package/xgboost/dask.py [hunk 1]
```diff
@@ -735,8 +735,9 @@ def train(client, params, dtrain, *args, evals=(), early_stopping_rounds=None,
     '''
     _assert_dask_support()
     client = _xgb_get_client(client)
+    global_config = config.get_config()
```
Comment (hcho3) [inline python-package/xgboost/dask.py:738]: Please collect all changes about global configuration and submit a separate pull request.

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: python-package/xgboost/dask.py (hunk 1 above)
Comment (trivialfis) [reply python-package/xgboost/dask.py:738]: It's WIP.  The fixed is already in master branch and I will remove it here.  I mentioned that I found the bug during debugging,  I was debugging this PR. ;-)

**Comment by trivialfis** [comment]
Closing to avoid spamming the list. Will reopen once ready.
